packages/upgrade/src/commands/migrate.js

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright IBM Corp. 2019, 2019
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import inquirer from 'inquirer';
+import { UpgradeError } from '../error';
+import { logger } from '../logger';
+import { Workspace, getAvailableWorkspaces } from '../workspace';
+
+export async function migrate(options, upgrades = []) {
+  logger.verbose('running migrate command with options: %o', options);
+
+  const migrations = upgrades
+    .filter((upgrade) => {
+      return upgrade.migrations && upgrade.migrations.length > 0;
+    })
+    .flatMap((upgrade) => {
+      return upgrade.migrations;
+    });
+
+  if (!migrations || migrations.length === 0) {
+    logger.info('No migrations available');
+    return;
+  }
+
+  // List available migrations
+  if (options.list) {
+    listMigrations(migrations);
+    return;
+  }
+
+  const workspaces = getAvailableWorkspaces(options.cwd);
+  if (workspaces.length === 0) {
+    throw new UpgradeError('Unable to find a workspace to migrate');
+  }
+
+  // Only run the migration specified in the options
+  if (options.migration) {
+    const migration = migrations.find((migration) => {
+      return migration.name === options.migration;
+    });
+
+    if (!migration) {
+      logger.error(
+        'Sorry, there is no migration with the name "%s"',
+        options.migration
+      );
+      listMigrations(migrations);
+    } else {
+      await runMigration(migration, workspaces, options);
+    }
+
+    return;
+  }
+
+  // Prompt for something to run when just 'migrate' is ran
+  const selectedMigrations = await getSelectedMigrations(migrations);
+  if (selectedMigrations.length === 0) {
+    logger.info('No migrations selected');
+    return;
+  }
+
+  if (selectedMigrations.length > 0) {
+    for (const migration of selectedMigrations) {
+      await runMigration(migration, workspaces, options);
+    }
+  }
+}
+
+/**
+ * @param {Array<Migrations>} migrations
+ * @returns {<void>}
+ */
+function listMigrations(migrations) {
+  logger.info('Available migrations:');
+  for (const migration of migrations) {
+    logger.info('%s (%s)', migration.name, migration.description);
```

Comment:
UX nit:

```suggestion
    logger.log('\t- %s (%s)', migration.name, migration.description);
```

Could help to have a plain list without the `[info]` preamble